lib/metadata/bucketfile/backend.js

```diff
@@ -29,152 +21,41 @@ class BucketFileInterface {
     constructor() {
         this.logger = logger;
         if (cluster.isMaster) {
-            this.startServer();
-        } else {
-            this.refcnt = 0;
-            this.waitreco = 0;
-            this.realReConnect();
-            this.notifier = new events.EventEmitter();
+            this.mdServer = new MetadataServer(
+            { metadataPath: METADATA_PATH,
+              metadataPort: METADATA_PORT,
+              log: config.log });
+            this.mdServer.startServer();
+        }
+        this.mdClient = new MetadataClient(
+            { metadataHost: 'localhost',
+              metadataPort: METADATA_PORT,
+              log: config.log });
+        this.mdDB = this.mdClient.openDB();
+        this.metastore = this.mdDB.openSub(METASTORE);
+        if (cluster.isMaster) {
```

Comment:
Now I remember why there are two `isMaster` checks done, it's because the setup has to be done in three phases:

1. start the metadata server, so that it can accept clients
2. create a metadata client and connect to the server
3. setup the basic dbs on the storage (namely the usersBucket db), which is done through the client (previously it was done on the local db, but it would not work with a remote db).

So steps 1 and 3 are done exclusively on the master.